authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/validator/AuthRequestValidator.java

```diff
@@ -172,20 +175,92 @@ protected void validateBiometricTimestamps(List<BioIdentityInfoDTO> biometrics,
 							new Object[] { String.format(BIO_PATH, i, IdAuthCommonConstants.DATA) },
 							IdAuthenticationErrorConstants.MISSING_INPUT_PARAMETER.getErrorMessage());
 				} else {
-					validateReqTime(bioIdentityInfoDTO.getData().getTimestamp(), errors,
-							String.format(BIO_PATH, i, DATA_TIMESTAMP), this::biometricTimestampParser);
-
-					if (!errors.hasErrors()) {
-						validateDigitalIdTimestamp(bioIdentityInfoDTO.getData().getDigitalId(), errors,
-								String.format(BIO_PATH, i, DIGITAL_ID));
-					}
+					validateBioTxnId(authTxnId, errors, i, bioIdentityInfoDTO.getData().getTransactionId());
+					validateBiometricTimestampAndDigitalIdTimestamp(biometrics.size() - 1, errors, i,
+							bioIdentityInfoDTO.getData());
+					validateSuccessiveBioSegmentTimestamp(biometrics, errors, i, bioIdentityInfoDTO);
 				}
 			}
 		} else {
 			errors.rejectValue(IdAuthCommonConstants.REQUEST,
-					IdAuthenticationErrorConstants.MISSING_INPUT_PARAMETER.getErrorCode(), new Object[] { "request/biometrics" },
+					IdAuthenticationErrorConstants.MISSING_INPUT_PARAMETER.getErrorCode(),
+					new Object[] { "request/biometrics" },
+					IdAuthenticationErrorConstants.MISSING_INPUT_PARAMETER.getErrorMessage());
+		}
+	}
+
+	private void validateSuccessiveBioSegmentTimestamp(List<BioIdentityInfoDTO> biometrics, Errors errors, int index,
+			BioIdentityInfoDTO bioIdentityInfoDTO) {
+		if (!errors.hasErrors() && index != 0) {
+			long bioTimestampDiffInSeconds = Duration.between(DateUtils.parseToLocalDateTime(bioIdentityInfoDTO.getData().getTimestamp()),
+					DateUtils.parseToLocalDateTime(biometrics.get(index - 1).getData().getTimestamp()))
+					.toSeconds();
+			long digitalIdTimestampDiffInSeconds = Duration.between(DateUtils.parseToLocalDateTime(bioIdentityInfoDTO.getData().getDigitalId().getDateTime()),
+					DateUtils.parseToLocalDateTime(biometrics.get(index - 1).getData().getDigitalId().getDateTime()))
+					.toSeconds();
+			Long allowedTimeDiffInSeconds = env.getProperty(IdAuthConfigKeyConstants.BIO_SEGMENT_TIME_DIFF_ALLOWED, Long.class, 120L);
+			if (bioTimestampDiffInSeconds > allowedTimeDiffInSeconds) {
+				mosipLogger.error(SESSION_ID, this.getClass().getSimpleName(), VALIDATE,
+						IdAuthenticationErrorConstants.INVALID_BIO_TIMESTAMP);
+				errors.rejectValue(IdAuthCommonConstants.REQUEST,
+						IdAuthenticationErrorConstants.INVALID_BIO_TIMESTAMP.getErrorCode(), new Object[] { allowedTimeDiffInSeconds },
+						IdAuthenticationErrorConstants.INVALID_BIO_TIMESTAMP.getErrorMessage());
+			}
+			if (digitalIdTimestampDiffInSeconds > allowedTimeDiffInSeconds) {

```

Comment:
Move the computation of long digitalIdTimestampDiffInSeconds just above this condition.

Reply:
done